Modules/Microsoft365DSC/DSCResources/MSFT_AADGroupOwnerConsentSettings/MSFT_AADGroupOwnerConsentSettings.psm1

```diff
@@ -0,0 +1,538 @@
+        }
+
+        if ($PSBoundParameters.ContainsKey('BlockUserConsentForRiskyApps'))
+        {
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='BlockUserConsentForRiskyApps'; Value=$BlockUserConsentForRiskyApps})
+        }
+        else
+        {
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='BlockUserConsentForRiskyApps';   Value=$currentInstance.BlockUserConsentForRiskyApps})
+        }
+
+        if ($PSBoundParameters.ContainsKey('EnableAdminConsentRequests'))
+        {
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='EnableAdminConsentRequests'; Value=$EnableAdminConsentRequests})
+        }
+        else
+        {
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='EnableAdminConsentRequests';   Value=$currentInstance.EnableAdminConsentRequests})
+        }
+
+        if ($EnableGroupSpecificConsent -and -not [string]::IsNullOrEmpty($ConstrainGroupSpecificConsentToMembersOfGroupName))
+        {
+            $constrainConsentGroupObj = Get-MgGroup -Filter "DisplayName eq '$ConstrainGroupSpecificConsentToMembersOfGroupName'"
+            if ($null -eq $constrainConsentGroupObj -or $constrainConsentGroupObj.securityEnabled -ne $true)
+            {
+                $message = "ConstrainGroupSpecificConsentToMembersOfGroupName '$ConstrainGroupSpecificConsentToMembersOfGroupName' does not exist or is not a security group"
+                Add-M365DscEvent -Message $message `
+                    -Source $($MyInvocation.MyCommand.Source) `
+                    -EntryType Error `
+                    -EventId 2 `
+                    -EventType Error `
+                    -TenantId $TenantId
+
+                throw $message
+            }
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='ConstrainGroupSpecificConsentToMembersOfGroupId'; Value=$constrainConsentGroupObj.Id})
+        }
+        else
+        {
+            $valuesParam += [Microsoft.Graph.PowerShell.Models.MicrosoftGraphSettingValue]::DeserializeFromPSObject([pscustomobject]@{Name='ConstrainGroupSpecificConsentToMembersOfGroupId'; Value=$null})
+        }
+
+
+        $BodyParam = @{
+            DisplayName = $templateSettings.DisplayName
+            Values      = $valuesParam
+        }
+
+        if ($currentInstance.Ensure -eq 'Absent')
+        {
+            Write-Verbose -Message "UNEXPECTED: *Creating* the Azure AD Group Consent Setting"
```

Comment:
Why is this Unexpected??

Reply:
Good point. It isn't. The groupsetting may not exist if default values are used